### PR TITLE
group output

### PR DESCRIPTION
**Pull request by ericsciple**
Collapse log output into logical groups for a nicer experience

### CODE REVIEW

File: src/git-directory-helper.ts [hunk 1]
```diff
@@ -38,6 +44,7 @@ export async function prepareExistingDirectory(
     }
 
     try {
+      core.startGroup('Removing previously created refs, to avoid conflicts')
```
Comment (thboop) [inline src/git-directory-helper.ts]: I think we need to keep mark a flag to indicate we have started a group but not ended it so we can error handle correctly on this try-catch

Comment (ericsciple) [reply src/git-directory-helper.ts]: Unclosed groups dont really hurt anything. Intentionally did not complicate the code to track.

Also note, can't have nested groups. Starting a new group automatically closes any open group.

Thoughts?

Comment (ericsciple) [reply src/git-directory-helper.ts]: Otherwise try/finally may make sense

**Review by Thefaceofasaba — COMMENTED** [review]
Here

**Pull request merged**